powerauth-data-adapter/src/main/java/io/getlime/security/powerauth/app/dataadapter/impl/service/DataAdapterService.java

```diff
@@ -190,6 +190,46 @@ public void formDataChangedNotification(String userId, String organizationId, Fo
         throw new IllegalStateException("Invalid change entity type: " + change.getType());
     }
 
+    @Override
+    public CreateImplicitLoginOperationResponse createImplicitLoginOperation(String clientId, String[] scopes) throws DataAdapterRemoteException {
+        try {
+            final TppAppDetailResponse appDetail = tppEngineService.fetchAppDetail(clientId);
+
+            // Make sure there is only one item in scopes
+            if (scopes == null || scopes.length != 1) {
+                return null;
```

Comment:
Returning `null` seems a bit weird. Why not throw the exception instead? Furthermore, the error reason should probably be logged.